repository-module/src/main/java/com/gmail/zagurskaya/online/cash/repository/impl/ReviewsRepositoryImpl.java

```diff
@@ -0,0 +1,114 @@
+package com.gmail.zagurskaya.online.cash.repository.impl;
+
+import com.gmail.zagurskaya.online.cash.repository.ReviewsRepository;
+import com.gmail.zagurskaya.online.cash.repository.exception.ReviewsRepositoryException;
+import com.gmail.zagurskaya.online.cash.repository.exception.RoleRepositoryException;
+import com.gmail.zagurskaya.online.cash.repository.model.Reviews;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class ReviewsRepositoryImpl implements ReviewsRepository {
+
+    private static Logger logger = LoggerFactory.getLogger(ReviewsRepositoryImpl.class);
+
+    @Override
+    public List<Reviews> getReviews(Connection connection) {
+        String sql = String.format(
+                "SELECT * FROM `reviews` WHERE`isNotOpen`= 0");
+        List<Reviews> reviewList = new ArrayList<>();
+        try (PreparedStatement prepared = connection.prepareStatement(sql)) {
+            ResultSet resultSet = prepared.executeQuery(sql);
+            return getReviewsFromResult(resultSet);
+        } catch (SQLException e) {
+            logger.error(e.getMessage(), e);
+            throw new ReviewsRepositoryException("Database exception during  getReviews ", e);
+        }
+    }
+
+    public Reviews create(Connection connection, Reviews reviews) {
+        String sql = "INSERT INTO `reviews`( date , reviewId, description, isNotOpen) " +
+                "VALUES (?,?,?,?)";
+        List<Reviews> reviewList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setDate(1, reviews.getDate());
+            preparedStatement.setLong(2, reviews.getUserId());
+            preparedStatement.setString(3, reviews.getDescription());
+            preparedStatement.setLong(7, reviews.getIsNotOpen() ? 1 : 0);
+            preparedStatement.executeUpdate();
+
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                reviews.setId(resultSet.getLong(1));
+            }
+            return reviews;
+        } catch (SQLException e) {
+            logger.error(e.getMessage(), e);
+            throw new ReviewsRepositoryException("Database exception during add review ", e);
+        }
+    }
+
+    @Override
+    public Reviews getReview(Connection connection, Long id) {
+        String sql = "SELECT * FROM reviews  WHERE id = ? ";
+        List<Reviews> reviewList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getReviewsFromResult(resultSet).get(0);
```

Comment:
Need to use separate method got getting ONE result. You can get Indexofboundexception.